TermTk/TTkWidgets/progressbar.py

```diff
@@ -0,0 +1,148 @@
+# MIT License
+#
+# Copyright (c) 2022 Eugenio Parodi <ceccopierangiolieugenio AT googlemail DOT com>
+#                    Luchr          <https://github.com/luchr>
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import math
+
+from TermTk.TTkCore.cfg       import TTkCfg
+from TermTk.TTkCore.constant  import TTkK
+from TermTk.TTkCore.string    import TTkString
+from TermTk.TTkCore.signal    import pyTTkSignal
+from TermTk.TTkWidgets.widget import TTkWidget
+from TermTk.TTkTemplates.lookandfeel import TTkLookAndFeel
+
+class TTkLookAndFeelPBar(TTkLookAndFeel):
+    def color(self, value, minimum, maximum):
+        return TTkCfg.theme.progresssBarColor
+    def text(self, value, minimum, maximum):
+        percent = round(100*(value-minimum)/(maximum-minimum))
+        return TTkString(f"{percent:3}%", color=TTkCfg.theme.progressBarTextColor)
+
+'''
+     Progressbar:  |████████▌      |
+        rest block          ^
+        full blocks ^^^^^^^^
+'''
+class TTkProgressBar(TTkWidget):
+    __slots__ = (
+        '_value', '_value_min', '_value_max', '_textWidth',
+        # Signals
+        'valueChanged')
+
+    def __init__(self, *args, **kwargs):
+        self.valueChanged = pyTTkSignal(float)
+        TTkWidget.__init__(self, *args, **kwargs)
+        self.setLookAndFeel(TTkLookAndFeelPBar())
+        self._textWidth = kwargs.get('textWidth', 4)
```

Comment:
Should textWidth be a property of the lookAndFeel object?
we can add 2 fields:

showText (boolean)
and textWidth

showTtext can be useful to enable/disable dynamically the text
in particular if you want to re-enable it without knowing the textwidth or just to use the last one.

Reply:
Nice. More and more is migrating into the look-and-feel.